packages/react-dom/src/__tests__/ReactDOMServerIntegrationLegacyContextDisabled-test.internal.js

```diff
@@ -104,7 +105,7 @@ describe('ReactDOMServerIntegrationLegacyContextDisabled', () => {
           <RegularFn />
         </span>
       </LegacyProvider>,
-      3,
+      render === clientRenderOnBadMarkup ? 6 : 3,
```

Comment:
In the new root we retry with client rendering on hydration mismatch. Since the warning isn't deduped, we get twice the errors.

With https://github.com/facebook/react/pull/30299 the warning is going to be deduped.